content/chat/typing.textile

```diff
@@ -5,3 +5,99 @@ product: chat
+
+```[realtime_javascript]
+    // Enter the chat participant into the presence set
+    channel.presence.enter((err) => {
+      // Custom functionality to update UI for chat participant that they are now online.
+    });
+```
+
+h3(#update). Update
+
+The chat participant status can be used as the indicator of whether they're currently typing or not, allowing other chat participants to see a visual cue. The status field is a text based field, so the value can be whichever text value of your choosing.
+
+```[realtime_javascript]
+    // Subscribe to any presence updates for the current presence set
+    channel.presence.subscribe('update', (memberUpdate) => {
+      // Update current array of users typing to display the one for memberUpdate as typing.
+    });
+```
+
+Depending on the UI, it is recommended to display a limited number of the names of chat participants typing. If that number is exceeded, it's common practice to instead change the visual indicator to a string such as "People are typing…".
+
+h3(#example). Demo example
+
+Below is an example of each part required for the typing indicator to be implemented. Once the client has an authToken with a clientId associated with it, you join the predefined channel and enter the presence set for that channel.
+
+The application needs to subscribe to any updates of that presence set and would then update the visual cues displaying which chat participants are currently typing.
+
+For the chat participant, an event listener waits for a keypress before publishing an update to the presence set, updating this chat participant's status with a string indicating they're typing. A timer is set to expire in five seconds to reset this status if there has been no further keys pressed in that time.
+
+```[realtime_javascript]
+    let isTyping = false;
+    let typingInterval;
+    let usersTyping = [];
+
+    var client = new Ably.Realtime(authToken);
+    var channel = client.channels.get('CHANNEL_NAME');
+
+    // Subscribe to any presence updates for the current presence set
+    channel.presence.subscribe('update', (memberUpdate) => {
+      // find your user by their client id
+      user = '';
+      // Adding user to array of users typing
+      usersTyping.push(user);
+    });
+
+    // Enter the chat participant into the presence set
+    channel.presence.enter((err) => {
+      // Custom functionality to update UI for chat participants that they are now online.
+    });
+
+    // Listener for when chat participant presses a key on their keyboard
+    document.addEventListener('keypress', (event) => {
```

Comment:
Are we able to do React examples? This could be really nice in a specific framework example :)

Reply:
I agree, further examples in React would be a great addition. But would it be ok to have a ticket to do this post release? I'm hopeful to not have any further delays on this docs.

Reply:
> Are we able to do React examples? This could be really nice in a specific framework example :)

I'm sorry, what do you mean when it comes to this specific eventlistener?

Reply:
Ah, I was just thinking that in most modern apps you wouldn't be using `document.xyz` because you'd be using a framework that hides that bit away, so thought having a React example might be more exemplary than plain JS :) But happy to defer React examples until later :)